### PR TITLE
Add Message with parsing/serialziation support

### LINKED ISSUE #15: Message envelopes
This isn't needed for TChannel, but Thrift actually wraps requests and responses in a message envelope with the following attributes:

* `name`: Name of the method being called.
* `type`: Type of the message. This is one of `call` (1), `reply` (2), `exception` (3), and `oneway` (4).
* `seqid`: A 32-bit integer used by the client to match responses to requests if they support interleaved requests and responses. The server's contract is to return the same seqid in the response that it got in the request.
* `body`: The actual request/response struct/union.

For binary protocol, that uses the following format:

    name~4 type:1 seqid:4 body

Where body is just laid out as-is.

We should implement some sort of API for this so that thriftrw can be used for HTTP clients as well.

Worth noting that the binary protocol also has a strict version of the message wrapper:

    versionAndType:4 name~4 seqid:4 body
    versionAndType = version:2 0x00 type:1
    version = 0x8001

For reads, we probably want to support both, the strict and non-strict version. For writes, we can probably just pick one and roll with it.

### PR DESCRIPTION
**Pull request by abhinav**
Add `<module>.dumps.message` and `<module>.loads.message` to serialize and deserialize requests/responses inside messages.

Messages are serialized into the non-strict format, but parsing of both, strict and non-strict formats is supported. For more information on what these words mean, see #15.

Also added a basic working example using HTTP.

Resolves #15.

### CODE REVIEW

**Comment by abhinav** [comment]
CC @breerly @blampe @junchaowu

File: README.rst [hunk 1]
```diff
@@ -112,12 +112,62 @@ You can use the library to send and receive requests and responses like so,
 
         return blog.dumps(response)
 
+Note that this example sends and receives just the request/response payload. It
+does not wrap the payload in a message envelope as expected by Apache Thrift.
+If you want to send or receive standard Apache Thrift requests to talk to other
```
Comment (blampe) [inline README.rst]: Add a header above this line for nicer linking pls :)

File: README.rst [hunk 2]
```diff
@@ -112,12 +112,62 @@ You can use the library to send and receive requests and responses like so,
...
+``dumps.message`` APIs. For example,
+
+.. code-block:: python
+
+    # client.py
+
+    def new_post():
+        post = blog.PostDetails(...)
+        request = BlogService.newPost.request(post)
+        payload = blog.dumps.message(request)
```
Comment (blampe) [inline README.rst]: It `message` HTTP-specific? Would `blog.dumps.http` (or `blog.http.dumps`) be more obvious?

File: README.rst [hunk 3]
```diff
@@ -112,12 +112,62 @@ You can use the library to send and receive requests and responses like so,
...
+
+    # server.py
+
+    def handle_request(request_payload):
+        message = blog.loads.message(BlogService, request_payload)
+        if message.name == 'newPost':
+            request = message.body
+            # ...
+            response = BlogService.newPost.response(success=post_uuid)
+            return blog.dumps.message(response, seqid=message.seqid)
```
Comment (blampe) [inline README.rst]: What's the impact of omitting `seqid` when we have thriftrw on the other end? What about Apache?

File: README.rst (hunk 2 above)
Comment (abhinav) [inline README.rst]: It's not HTTP specific. Even Thrift over TCP or TMemoryBuffer would use this. It's more of a TProtocol thing. All Thrift protocols define a message envelope.

File: README.rst (hunk 3 above)
Comment (abhinav) [inline README.rst]: The client may or may not care about the seqid. Depends on whether they expect to receive out of order responses. If the client is synchronous, they don't care. If the client is asynchronous, they're most likely using a message-id like mechanism to match responses up to ongoing requests. In that case they do care about the seqid.

The contract on the server side is: Responses for requests must have the same seqid as their corresponding requests. The server must follow this contract.

File: README.rst (hunk 1 above)
Comment (abhinav) [inline README.rst]: Done

**Comment by blampe** [comment]
:sheep:

**Pull request merged**